docs/design/ocs-to-ocs-api.md

```diff
@@ -0,0 +1,96 @@
+# FEATURE: OCS to OCS API changes
+
+## Summary
+Allow customers to share OCS storage across multiple OCP clusters. Each
+consumer OCP cluster will have OCS deployed that connects to the provider
+cluster.
+
+## Motivation
+This feature gives flexibility to customers and lets them share storage and
+better utilize their OpenShift clusters.
+
+### Goals
+Allow a customer to have a single OCS cluster sharing the storage across
+multiple OCP clusters. Each consumer OCP cluster will have OCS deployed in
+external mode that connects to an internal mode provider cluster. Also, allow
+us to reduce the cost of storage and maintenance.
+
+### Non-Goals
+Networking b/w two OCP clusters.
+
+## Proposal
+Introduce additional StorageCluster spec fields to support a new external mode
+where the storage is an OCS cluster.
+
+## Design Details
+
+### API changes required for a provider deployment
+```yaml
+spec:
+    allowRemoteStorageConsumers: <bool>
+```
+
+* `spec.allowRemoteStorageConsumers` Indicates that the OCS cluster should
+deploy the needed components to enable connections from remote consumers.
+When enabled the operator will deploy an API server that will allow onboarding,
+offboarding and second-day operation requests coming from consumer clusters.
+
+- onboarding: Configure the consumer to use the storage provided by the
+provider cluster. The consumer will install the OCS in external mode and
+connect to the provider cluster.
+- offboarding: Uninstall the ocs cluster and disconnect it from the provider.
+
+### API changes required for a consumer deployment
+```yaml
+spec:
+    externalStorage:
+        enable: <bool> (existing field)
+        storageProviderKind: <string> (ocs/rhcs) (default:rhcs)
+        connectionString: <string>
```

Comment:
How to handle StorageConsumer CR UUID?  Based on the `Revised Consumer Onboarding Proposal`,  storageConsumer CR won't be created in advance. So `connectionString` might not have any UUID.   Do we need a separate field for the UUID ?